cachito/workers/prune_archives.py

```diff
@@ -0,0 +1,233 @@
+            yield _ParsedArchive.from_path(path)
+        else:
+            log.debug("%s does not appear to be a source archive.", path)
+
+
+def _resolve_source_archive(parsed_archive: _ParsedArchive) -> Optional[_ResolvedArchive]:
+    """Return a _ResolvedArchive if a matching request is found via the API."""
+    latest_request = _get_latest_request(parsed_archive)
+    if latest_request is None:
+        log.debug("Archive %s could not be resolved via the API.", parsed_archive.path)
+        return None
+
+    return _ResolvedArchive(
+        parsed_archive.path,
+        datetime.strptime(latest_request["created"], "%Y-%m-%dT%H:%M:%S.%f").replace(
+            tzinfo=timezone.utc
+        ),
+        latest_request["id"],
+    )
+
+
+def _get_stale_archives(
+    older_than: datetime, api_calls_per_second: int
+) -> Generator[_ResolvedArchive, None, None]:
+    """
+    Return a Generator of _ResolvedArchives that are all stale.
+
+    The API requests are ratelimited to prevent potentially overwhelming the API
+    with a background maintenance task.
+    """
+
+    @sleep_and_retry
+    @limits(calls=api_calls_per_second, period=1)
+    def resolve_source_archive_ratelimited(archive: _ParsedArchive) -> Optional[_ResolvedArchive]:
+        return _resolve_source_archive(archive)
+
+    for parsed_archive in _get_parsed_source_archives(ARCHIVE_DIR):
+        resolved_archive = resolve_source_archive_ratelimited(parsed_archive)
+        if resolved_archive and resolved_archive.created < older_than:
+            yield resolved_archive
+
+
+def _process_stale_archives(
+    older_than: datetime,
+    api_calls_per_second: int,
+    delete: bool = False,
+    limit: Optional[int] = None,
+) -> None:
+    """List stale source archives up to the limit, optionally deleting them."""
+    for archive in islice(_get_stale_archives(older_than, api_calls_per_second), limit):
+        log.info(
+            f"Archive {archive.path} is stale. The most recent request_id="
+            f"{archive.latest_request_id} at {archive.created}"
+        )
+        if delete:
+            log.info(f"Deleting {archive.path}")
+            archive.path.unlink()
+
+
+def _validate_older_than(older_than: Optional[datetime]) -> datetime:
+    """Ensure that the value of the --older-than CLI option is not more recent than the minimum."""
+    older_than_utc = (
+        DEFAULT_AGE_DATETIME if older_than is None else older_than.replace(tzinfo=timezone.utc)
+    )
+    if older_than_utc > MINIMUM_AGE_DATETIME:
+        raise typer.BadParameter(f"cannot be more recent than {MINIMUM_AGE_DATETIME}")
+    return older_than_utc
+
+
+@app.command("delete")
+def delete_archives(
+    older_than: Annotated[
+        Optional[datetime],
+        typer.Option(
+            callback=_validate_older_than,
+            formats=["%Y-%m-%d"],
+            help="Deletes archives that are older than the specified date (UTC). YYYY-MM-DD",
```

Comment:
**nitpick:** 
> "Deletes archives that are older than the specified date (UTC)

1.  We should never ask end users to enter anything in UTC, because nobody wants to do the calculation in their head, we need to do the conversions automatically
2. mentioning UTC in context of plain dates might lead to the following question "Uhm, so I want to prune everything older than <date> in my timezone, but <date> is always `T00:00` which means that the UTC date would look differently, so which value do I put in there???" I mean this is pure bikeshedding and for the purpose of this script it doesn't matter at all, just something to be aware of.
3. using `replace()` on `datetime` instances doesn't adjust the time values underneath [1], it just forces a new field to the object meaning that you may have filtered out some values out of the evaluation, but here again, not a big deal in this particular case, it just brings questions to however reads the code.

For the sake of argument what I would suggest is to drop the UTC part from the help message, treat all inputs as local timezone to a particular system and then do
```
datetime.datetime.strptime("<timespec>", <format>).astimezone(datetime.timezone.utc)
```
in all related places to make sure we represent and base everything correctly.

[1] https://docs.python.org/3/library/datetime.html#datetime.datetime.replace

Reply:
Fixed :+1: 

User inputs should now be localtime and are converted to UTC